Bottom: ac8a41bda9fb41deb183b8da91124b2b0d64c756
Top:    6d624e1c7db6382be72025ed20dbbae6496b5dc9
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-14 16:24:52 -0500

Support asset host


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index cfb91cca..1a6bd7d1 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -9,6 +9,11 @@
     - buildbot
     - buildbot-master
   tasks:
+  - add_host:
+      name: nexus # required. The hostname/ip of the host to add to the inventory, can include a colon and a port number.
+      groups: services # not required. The groups to add the hostname to, comma separated.
+      ip_addr: "{{ nexus_ip_addr }}"
+      uri: "http://{{nexus_hostname}}/repository"
   - name: Scan Gitlab ssh keys
     command: ssh-keyscan {{gitlab_hostname}}
     register: gitlab_keys
@@ -35,7 +40,10 @@
     - buildbot
     - buildbot-worker
   vars:
+    asset_host: "{{hostvars['nexus']['uri']}}/download"
     buildbot_master: "{{ hostvars['buildbot-master'].ip_addr }}"
+    container_etc_hosts:
+      nexus.novatech-llc.com: "{{hostvars['nexus']['ip_addr']}}"
     known_hosts: "{{gitlab_keys.stdout}}"
   tasks:
   - name: Scan Gitlab ssh keys
diff --git a/docker/buildbot-worker-ntel/Dockerfile b/docker/buildbot-worker-ntel/Dockerfile
index 729332d4..5d040ea8 100644
--- a/docker/buildbot-worker-ntel/Dockerfile
+++ b/docker/buildbot-worker-ntel/Dockerfile
@@ -68,8 +68,9 @@ RUN wget https://bootstrap.pypa.io/get-pip.py \
 
 # Install software for image signing
 ARG PTK_HSM_HOST="127.0.0.1"
+ARG PTK_URI="http://127.0.0.1/610-009981-015_SW_PTK_5.3_Client_RevA.tar"
 
-RUN curl http://172.16.64.2/~georgem/610-009981-015_SW_PTK_5.3_Client_RevA.tar \
+RUN curl ${PTK_URI} \
 |   tar -v -x -C /tmp \
 &&  alien --to-deb --install --scripts /tmp/610-009981-015_SW_PTK_5.3_Client_RevA/SDKs/Linux64/ptkc_sdk/PTKcpsdk-5.3.0-16.x86_64.rpm \
 &&  alien --to-deb --install --scripts /tmp/610-009981-015_SW_PTK_5.3_Client_RevA/SDKs/Linux64/ptkc_runtime/PTKcprt-5.3.0-15.x86_64.rpm \
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git b/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git
index 7dbac6ef..b5c0c70e 100644
--- a/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git
+++ b/docker/buildbot-worker-ptxdist/buildbot/.ptxdist/ptxdistrc-git
@@ -36,12 +36,12 @@ PTXCONF_SETUP_SRCDIR="/cache/downloads"
 #
 # PTXCONF_SETUP_NO_DOWNLOAD is not set
 # PTXCONF_SETUP_PTXMIRROR_ONLY is not set
-PTXCONF_SETUP_PTXMIRROR="http://www.pengutronix.de/software/ptxdist/temporary-src"
-PTXCONF_SETUP_DEBMIRROR="http://ftp.uni-kl.de/debian"
-PTXCONF_SETUP_SFMIRROR="http://downloads.sourceforge.net/sourceforge"
-PTXCONF_SETUP_GNUMIRROR="http://ftp.uni-kl.de/pub/gnu"
-PTXCONF_SETUP_XORGMIRROR="http://xorg.mirrors.pair.com/pub/X11/ftp.x.org"
-PTXCONF_SETUP_KERNELMIRROR="http://www.kernel.org/pub/linux"
+PTXCONF_SETUP_PTXMIRROR="http://nexus.novatech-llc.com/repository/orphans"
+PTXCONF_SETUP_DEBMIRROR="http://nexus.novatech-llc.com/repository/debian"
+PTXCONF_SETUP_SFMIRROR="http://nexus.novatech-llc.com/repository/sourceforge"
+PTXCONF_SETUP_GNUMIRROR="http://nexus.novatech-llc.com/repository/gnu"
+PTXCONF_SETUP_XORGMIRROR="http://nexus.novatech-llc.com/repository/xorg"
+PTXCONF_SETUP_KERNELMIRROR="http://nexus.novatech-llc.com/repository/kernel"
 # PTXCONF_SETUP_CHECK_ALWAYS is not set
 PTXCONF_SETUP_CHECK_NOTEMPTY=y
 # PTXCONF_SETUP_CHECK_NEVER is not set
diff --git a/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg b/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
index 8dc2a29b..ee683c3d 100644
--- a/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
+++ b/docker/buildbot-worker-ptxdist/buildbot/.pydistutils.cfg
@@ -1,2 +1,2 @@
 [easy_install]
-index-url=http://172.16.103.72/repository/PyPi_Proxy/simple
+index-url=http://nexus.novatech-llc.com/repository/pypi/simple
diff --git a/roles/buildbot-worker-ntel/defaults/main.yml b/roles/buildbot-worker-ntel/defaults/main.yml
index afd6f6b3..4f51008e 100644
--- a/roles/buildbot-worker-ntel/defaults/main.yml
+++ b/roles/buildbot-worker-ntel/defaults/main.yml
@@ -33,6 +33,9 @@ container_env:
   BUILDBOT_WORKER_NAME: 'worker-ntel'
   BUILDBOT_WORKER_PASS: 'pass'
 
+# extra hosts to add to container's /etc/hosts file
+container_etc_hosts: []
+
 # the name of the container being started
 container_name: '{{ docker_name_prefix }}{{ hostname }}'
 
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
index f60ef127..e4ae40b0 100644
--- a/roles/buildbot-worker-ntel/tasks/main.yml
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -72,10 +72,12 @@
     tag: '{{ docker_image_tag }}'
     path: '{{ image_dir }}'
     buildargs: '{{ image_args }}'
+    extra_hosts: '{{ container_etc_hosts }}'
     force: true
 
 - name: Start buildbot container
   docker_container:
+    etc_hosts: '{{ container_etc_hosts }}'
     name: '{{ container_name }}'
     image: '{{ image_name }}:{{ docker_image_tag }}'
     networks: '{{ container_networks }}'
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index 3dc6fe6c..3fda3523 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -36,6 +36,9 @@ container_env:
 # the name of the container being started
 container_name: '{{ docker_name_prefix }}{{ hostname }}'
 
+# network configuration for the confainer
+container_networks: []
+
 # port configuration of the container
 container_port_args: []
 
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index ce43487c..dada6fd4 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -75,6 +75,7 @@
     tag: '{{ docker_image_tag }}'
     path: '{{ image_dir }}'
     buildargs: '{{ image_args }}'
+    extra_hosts: '{{ container_etc_hosts }}'
     force: true
 
 - name: Start buildbot container
diff --git a/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2 b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
index f9fd2372..38e82192 100644
--- a/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
+++ b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
@@ -1,2 +1,2 @@
 [easy_install]
-index_url = {{hostvars['buildbot-worker-devpi'].uri}}
+index_url = {{hostvars['nexus'].uri}}/pypi/simple
